src/app/core/services/phase.service.ts

```diff
@@ -200,16 +204,22 @@ export class PhaseService {
           return this.verifySessionAvailability(this.sessionData);
         }
       }),
-      flatMap((isSessionCreated: boolean) => {
-        if (!isSessionCreated) {
-          throw new Error('Session Availability Fix failed.');
-        }
-        return this.labelService.synchronizeRemoteLabels();
-      }),
+      this.checkSessionCreation(),
+      this.labelService.syncLabels(),
       retry(1)  // Retry once, to handle edge case where GitHub API cannot immediately confirm existence of the newly created repo.
     );
   }
 
+  public checkSessionCreation(): UnaryFunction<Observable<boolean>, Observable<boolean>> {
+    return pipe(
+      throwIfFalse(
+        (isSessionCreated: boolean) => isSessionCreated,
+        () => new Error(SESSION_AVALIABILITY_FIX_FAILED)
+      ),
+      map((isSessionCreated: boolean) => isSessionCreated)
```

Comment:
I don't think this `map` is needed, any reason for adding this?

Reply:
ah I wanted to be a bit more explicit that we are returning the original boolean observable. 

But I don't think we should need it as long we define the return type properly